tests/wpt/mozilla/tests/mozilla/custom_auto_rooter.html

```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<html>
+<meta charset="utf-8">
+<title>Types using Custom Auto Rooter work correctly</title>
+<head>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+</head>
+<script>
+test(function() {
+  function assert_array_equals(lhs, rhs) {
+    assert_equals(lhs.length, rhs.length);
+    for (var i = 0; i < lhs.length; i++) {
+      assert_equals(lhs[i], rhs[i]);
+    }
+  }
+
+  var t = new TestBinding;
+
+  var seq = [5, "str", { key: "val" }, undefined];
+  assert_array_equals(seq, t.anySequencePassthrough(seq));
+
+  var seq = new Array(null, {'inner': 1, 'a': 'as'}, {'inner': 2}, null, undefined);
+  assert_array_equals(seq, t.anySequencePassthrough(seq));
+
+  var seq = new Array(null, {'inner': 1, 'a': 'as', 'toString': function() { seq = null; return Object.prototype.toString(this); }}, {'inner': 2}, null, undefined);
+  assert_array_equals(seq, t.anySequencePassthrough(seq));
```

Comment:
Just to verify, have you checked if any of these tests fail without your codegen changes?

Reply:
Yes. Previously, with https://github.com/Xanewok/servo/commit/01d1039311fd89cec84d6080333e32114fe03f80:
```ts
var seq = [5, "str", { key: "val" }, undefined];
console.log("seq: " + seq);
var lvalue = t.anySequencePassthrough(seq);
console.log("lvalue: " + lvalue);
console.log("rvalue: " + t.anySequencePassthrough(seq));
assert_array_equals(seq, t.anySequencePassthrough(seq), "Returned simple array is the same");

```
errors with:
```
 0:13.20 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:27314) "seq: 5,str,[object Object],"   
 0:13.20 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:27314) "lvalue: 0,5e-324,5e-324,5e-324"
 0:13.20 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:27314) "rvalue: 0,5e-324,5e-324,5e-324"
...
Expected PASS, got FAIL                                 
assert_array_equals: Returned simple array is the same property 0, expected 0 but got 5
```

Reply:
\o/